nncf/telemetry/extractors.py

```diff
@@ -0,0 +1,57 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from abc import ABC
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+
+SerializableData = str
```

Comment:
The `target_device` parameter is `Enum`.  You pass value from this enum directly. Does it work?

Reply:
Good catch, extended the default behaviour to automatically serialize `Enum`s by their `.value`.